pandas/types/cast.py

```diff
@@ -89,7 +89,8 @@ def trans(x):  # noqa
 
         if issubclass(dtype.type, np.floating):
             return result.astype(dtype)
-        elif dtype == np.bool_ or issubclass(dtype.type, np.integer):
+        elif (dtype == np.bool_ or issubclass(dtype.type, np.integer) and
```

Comment:
The issue in this case was exactly something like that: a timedelta dtype passed the `issubclass(dtype.type, np.integer)` check, and for this reason the specific `elif` close for datetimes and timedeltas more below at line 128 was not used for timedeltas (but was correct for datetimes).